java/src/jmri/jmrit/beantable/BeanTableDataModel.java

```diff
@@ -940,6 +943,17 @@ public Void doInBackground() {
             return null;
         }
 
+        /**
+         * Minimal implementation to catch and log errors
+         */
+        @Override
+        protected void done() {
+            try {
+                get();  // called to get errors
+            } catch (InterruptedException | java.util.concurrent.ExecutionException e) {
+                log.error("Exception while deleting bean", e);
+            }
+        }
```

Comment:
Shouldn’t InterruptedExceptions be rethrown?

Reply:
> Shouldn’t InterruptedExceptions be rethrown?

There's conflicting advise on that.  

An InterruptedException here was an interrupt to the SwingWorker thread (SwingWorker operations are synchronized with the Swing/AWT thread, but don't actually run on it; they use a custom thread pool) which has ceased this operation by the time `done()` is invoked. In that sense, the interrupt has _already_ done the right thing.

I haven't had time to think through what JMRI should generally do in this case.